### PR TITLE
♻️ Rebrand: Inference API -> Inference Endpoints (serverless)

### PR DESCRIPTION
**Pull request by SBrandeis**
cc @julien-c @gary149 @Pierrci @osanseviero 

The Inference API is changing names and becoming part of the Inference Endpoints product (`Inference Endpoints (serverless)` )

This PR renames all references to `Inference API` as `Inference Endpoints (serverless)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,12 +37,12 @@ await inference.textToImage({
 
 This is a collection of JS libraries to interact with the Hugging Face API, with TS types included.
 
-- [@huggingface/inference](packages/inference/README.md): Use the Inference API to make calls to 100,000+ Machine Learning models, or your own [inference endpoints](https://hf.co/docs/inference-endpoints/)!
+- [@huggingface/inference](packages/inference/README.md): Use Inference Endpoints to make calls to 100,000+ Machine Learning models
```
Comment (julien-c) [inline README.md:40]:
```suggestion
- [@huggingface/inference](packages/inference/README.md): Use Inference Endpoints (serverless) to make calls to 100,000+ Machine Learning models
```

File: packages/inference/README.md [hunk 2]
```diff
@@ -1,10 +1,11 @@
-# 🤗 Hugging Face Inference API
+# 🤗 Hugging Face Inference Endpoints
 
-A Typescript powered wrapper for the Hugging Face Inference API. Learn more about the Inference API at [Hugging Face](https://huggingface.co/docs/api-inference/index). It also works with [Inference Endpoints](https://huggingface.co/docs/inference-endpoints/index).
+A Typescript powered wrapper for the Hugging Face Inference Endpoints API. Learn more about Inference Endpoints at [Hugging Face](https://huggingface.co/inference-endpoints).
+It works wither both [serverless](https://huggingface.co/docs/api-inference/index) and [dedicated](https://huggingface.co/docs/inference-endpoints/index) Endpoints.
```
Comment (julien-c) [inline packages/inference/README.md:4]: 👍

File: packages/widgets/src/lib/components/InferenceWidget/shared/WidgetHeader/WidgetHeader.svelte [hunk 3]
```diff
@@ -53,7 +53,7 @@
 			<div class="flex items-center text-lg">
 				{#if !isDisabled}
 					<IconLightning classNames="-ml-1 mr-1 text-yellow-500" />
-					Inference API
+					Inference Endpoints (serverless)
```
Comment (julien-c) [inline packages/widgets/src/lib/components/InferenceWidget/shared/WidgetHeader/WidgetHeader.svelte:56]: 👍

**Review by julien-c — APPROVED** [review]
ok for me

File: packages/agents/README.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # 🤗 Hugging Face Agents.js
 
-A way to call Hugging Face models and inference APIs from natural language, using an LLM.
+A way to call Hugging Face models and inference Endpoints from natural language, using an LLM.
```
Comment (Pierrci) [inline packages/agents/README.md:3]:
```suggestion
A way to call Hugging Face models and Inference Endpoints from natural language, using an LLM.
```

File: packages/inference/README.md (hunk 2 above)
Comment (Pierrci) [reply packages/inference/README.md:4]:
```suggestion
It works with both [serverless](https://huggingface.co/docs/api-inference/index) and [dedicated](https://huggingface.co/docs/inference-endpoints/index) Endpoints.
```

File: README.md [hunk 5]
```diff
@@ -37,12 +37,12 @@ await inference.textToImage({
...
 This is a collection of JS libraries to interact with the Hugging Face API, with TS types included.
 
-- [@huggingface/inference](packages/inference/README.md): Use the Inference API to make calls to 100,000+ Machine Learning models, or your own [inference endpoints](https://hf.co/docs/inference-endpoints/)!
+- [@huggingface/inference](packages/inference/README.md): Use Inference Endpoints (serverless) to make calls to 100,000+ Machine Learning models
 - [@huggingface/hub](packages/hub/README.md): Interact with huggingface.co to create or delete repos and commit / download files
 - [@huggingface/agents](packages/agents/README.md): Interact with HF models through a natural language interface
 
 
-With more to come, like `@huggingface/endpoints` to manage your HF Endpoints!
+With more to come, like `@huggingface/endpoints` to manage your dedicated Inference Endpoints!
```
Comment (Pierrci) [inline README.md:46]:
```suggestion
```
From the updated readme, the package is supposed to work with both serverless and dedicated flavors, so I would remove this, no?

**Review by Pierrci — APPROVED** [review]

File: packages/tasks/src/model-data.ts [hunk 6]
```diff
@@ -78,7 +78,7 @@ export interface ModelData {
 	 */
 	widgetData?: WidgetExample[] | undefined;
 	/**
-	 * Parameters that will be used by the widget when calling Inference API
+	 * Parameters that will be used by the widget when calling Inference Endpoints (serverless)
```
Comment (osanseviero) [inline packages/tasks/src/model-data.ts:81]: Sometimes we use `Inference Endpoints (serverless)` and sometimes `Serverless Inference Endpoints`. Let's try to be consistent

**Review by osanseviero — APPROVED** [review]
Very nice!

**Pull request merged**